### PR TITLE
Avoid passing incomplete data to perflab_render_plugin_card() and show error when plugin directory API query fails

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

When working on Image Prioritizer (#1172), I added the plugin to the `perflab_get_standalone_plugin_data()` but then I started getting PHPUnit test failures. I also got a bunch of warnings on the Performance screen:

> image

The issue here is that Image Prioritizer is not on WordPress.org yet, so the call to `perflab_query_plugin_info()` is returning an empty array. Nevertheless, this logic in `perflab_render_plugin_card()` is not working as expected:

https://github.com/WordPress/performance/blob/cb5a37e8871e06564ac23e793b2121d6181dc0b8/includes/admin/plugins.php#L130-L134

This is because we are merging the response from WordPress.org with our own data from `perflab_get_standalone_plugin_data()` meaning `$plugin_data` will _never_ be empty:

https://github.com/WordPress/performance/blob/cb5a37e8871e06564ac23e793b2121d6181dc0b8/includes/admin/plugins.php#L76-L88

This scenario could happen not only if the plugin is not on WordPress.org but also if the query fails due to WordPress.org being down. So with this PR an error is now shown when an error occurs:

image

This PR also removes the standalone plugin version from being referenced on the Performance screen. It was only referenced in `aria-label` and `data-title` attributes.

Lastly, when a WordPress.org Plugins API query fails during installation, an error message is now displayed rather than just a generic error.

Related: #1170

### CODE REVIEW

File: includes/admin/plugins.php [hunk 1]
```diff
@@ -125,22 +151,15 @@ function perflab_render_plugins_ui() {
...
-	}
 
-	// Remove any HTML from the description.
+	$name        = wp_strip_all_tags( $plugin_data['name'] );
 	$description = wp_strip_all_tags( $plugin_data['short_description'] );
-	$title       = $plugin_data['name'];
 
 	/** This filter is documented in wp-admin/includes/class-wp-plugin-install-list-table.php */
 	$description = apply_filters( 'plugin_install_description', $description, $plugin_data );
-	$version     = $plugin_data['version'];
```
Comment (westonruter) [inline includes/admin/plugins.php:142]: I didn't see a need for the version to be mentioned among the features, so I removed it.

File: includes/admin/load.php [hunk 2]
```diff
@@ -260,14 +260,14 @@ function perflab_install_activate_plugin_callback() {
...
-		if ( ! $api ) {
+		if ( $api instanceof WP_Error ) {
 			wp_die(
 				wp_kses(
 					sprintf(
 						/* translators: %s: Support forums URL. */
 						__( 'An unexpected error occurred. Something may be wrong with WordPress.org or this server&#8217;s configuration. If you continue to have problems, please try the <a href="%s">support forums</a>.', 'default' ),
 						__( 'https://wordpress.org/support/forums/', 'default' )
-					),
+					) . ' ' . $api->get_error_message(),
```
Comment (westonruter) [inline includes/admin/load.php:278]: Now the specific error is shown rather than just the generic message.

File: includes/admin/load.php [hunk 3]
```diff
@@ -304,7 +304,7 @@ function perflab_install_activate_plugin_callback() {
 
 	$result = activate_plugin( $plugin_basename );
 	if ( is_wp_error( $result ) ) {
-		wp_die( esc_html( $result->get_error_message() ) );
+		wp_die( wp_kses_post( $result->get_error_message() ) );
```
Comment (westonruter) [inline includes/admin/load.php:313]: This fixes HTML mistakenly being escaped:

image

Now:

image

File: includes/admin/plugins.php [hunk 4]
```diff
@@ -74,20 +88,41 @@ function perflab_render_plugins_ui() {
...
+
+		// Skip if the plugin is not on WordPress.org or there was a network error.
+		if ( $api_data instanceof WP_Error ) {
+			wp_admin_notice(
+				esc_html(
+					sprintf(
+						/* translators: %s is the plugin slug */
+						__( 'Failed to query WordPress.org Plugin Directory for plugin "%s".', 'performance-lab' ) . ' %s',
+						$plugin_slug,
+						$api_data->get_error_message()
```
Comment (swissspidy) [inline includes/admin/plugins.php]:
```suggestion
						/* translators: 1: plugin slug. 2: error message. */
						__( 'Failed to query WordPress.org Plugin Directory for plugin "%1$s". %2$s', 'performance-lab' ),
						$plugin_slug,
						$api_data->get_error_message()
```

Comment (westonruter) [reply includes/admin/plugins.php]: Actually I did it this way to reuse the core translation string.

Comment (westonruter) [reply includes/admin/plugins.php]: Oh no, that was somewhere else I guess.

**Review by swissspidy — APPROVED** [review]

File: includes/admin/plugins.php (hunk 4 above)
Comment (westonruter) [reply includes/admin/plugins.php]: Oh yeah, it should do the same thing as I commented at https://github.com/WordPress/performance/pull/1175/files#r1576999210

Comment (westonruter) [reply includes/admin/plugins.php]: Meh. That's too generic. Nevermind.

File: includes/admin/plugins.php [hunk 5]
```diff
@@ -24,25 +24,39 @@ function perflab_query_plugin_info( string $plugin_slug ) {
...
+
 	$plugin = plugins_api(
 		'plugin_information',
 		array(
 			'slug'   => $plugin_slug,
-			'fields' => array(
-				'short_description' => true,
-				'icons'             => true,
-			),
+			'fields' => array_fill_keys( $fields, true ),
```
Comment (mukeshpanchal27) [inline includes/admin/plugins.php:42]: Nice ❤️

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**